app/login/page.test.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import Login from './page';
+
+const mockLoginAccount = jest.fn();
+const mockPush = jest.fn();
+
+let emailInput: HTMLElement,
+  passwordInput: HTMLElement,
+  continueButton: HTMLElement;
+
+const mockUseAuthContext = {
+  loginAccount: mockLoginAccount,
+  isSignedIn: false,
+};
+
+jest.mock('next/navigation', () => ({
+  useRouter() {
+    return {
+      push: mockPush,
+    };
+  },
+}));
+
+jest.mock('../../context/AuthContextProvider', () => ({
+  useAuthContext() {
+    return {
+      ...mockUseAuthContext,
+    };
+  },
+}));
+
+describe('Login', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    render(<Login />);
+    emailInput = screen.getByTestId('email');
+    passwordInput = screen.getByTestId('password');
+    continueButton = screen.getByTestId('continue-button');
+  });
+  test('should render the login page', () => {
+    expect(emailInput).toBeInTheDocument();
+    expect(passwordInput).toBeInTheDocument();
+    expect(continueButton).toBeInTheDocument();
+  });
+
+  test('should update email state when input value changes', () => {
+    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
```

Comment:
Q: I wonder if you could store your test values like `test@example.com` and `pasword123` into an object, and then call that object's properties?  This way if you want/have to change the test values at any point, it will be more convenient, faster and efficient to do so?

Reply:
That's a good call; I'll give it a shot and if if it passes will add it in the next commit.

Reply:
This was added in my latest commit. Good idea!